server/controllers/payment.js

```diff
@@ -0,0 +1,53 @@
+const Payment = require("../models/Payment");
+const stripe = require("../utils/stripe");
+const asyncHandler = require("express-async-handler");
+const stripe = require("stripe")(
+  "sk_test_51KcFwoAZ9gJcnOgDUNatBvPLDR4J7tfHodr55G24tP3UZgO3nyWtvE272SaqYqNHtpf22UTCUX2hDM9s0XXW4W5X0064ASBu8x"
+);
+
+// @route GET /payment
+// @desc Get all payments for the signed user
+// @access Private
+exports.getAllPayments = asyncHandler(async (req, res, next) => {
+  const payments = await Payment.find({ userId: req.params.id });
+  res.send(payments);
+});
+
+// @route GET /payments/:paymentid
+// @desc Get a payment
+// @access Private
+exports.getAllPayments = asyncHandler(async (req, res, next) => {
+  const charge = await stripe.charges.retrieve("ch_3KcGB2AZ9gJcnOgD3mFC1zrY");
```

Comment:
Hard coded id here, we should actually be retrieving the payments from the payment in our database rather directly from Stripe.